velox/expression/tests/ExpressionFuzzer.cpp

```diff
@@ -536,6 +537,30 @@ std::vector<core::TypedExprPtr> ExpressionFuzzer::generateRegexpReplaceArgs(
   return inputExpressions;
 }
 
+std::vector<core::TypedExprPtr> ExpressionFuzzer::generateSwitchArgs(
+    const CallableSignature& input) {
+  VELOX_CHECK_EQ(
+      input.args.size(),
+      2,
+      "Only two inputs are expected from the template signature.");
+  size_t cases = boost::random::uniform_int_distribution<uint32_t>(1, 5)(rng_);
+  bool useFinalElse =
+      boost::random::uniform_int_distribution<uint32_t>(0, 1)(rng_) > 0;
+
+  auto conditionClauseType = input.args[0];
+  auto thenClauseType = input.args[1];
```

Comment:
What about making thenClauseType be input.returnType? Then the signature of switch expression can be boolean -> T.

Reply:
I was only retaining both inputs types for readability so that the inputs match 1-1 with our understanding of the idea of a switch expression (switch (condition, then): boolean, T -> T). If we want to go leaner we can also remove any input params completely since we know the condition clause will always be boolean and the then/else will always be the return type. I dont have a strong preference either way, so please let me know what you think.

Reply:
I think it's OK to remove the input parameter type. Alternatively, we should change it to something like `boolean, K -> T` because we don't require the input type of then-clause to be the same as their output type.

Reply:
As discussed offline with @kagamiori, I'll add a comment describing the notation (switch (condition, then): boolean, T -> T) to improve readability.